eng/scripts/Language-Settings.ps1

```diff
@@ -118,6 +118,30 @@ function Get-AllPackageInfoFromRepo ([string] $ServiceDirectory) {
   return $allPackageProps
 }
 
+function Get-rust-AdditionalValidationPackagesFromPackageSet ($packagesWithChanges, $diff, $allPackageProperties) {
+  $additionalPackages = @()
+
+  # if the change was in a service directory, but not in a package directory, test all the packages in the service directory
+  [array]$serviceFiles = ($diff.ChangedFiles + $diff.DeletedFiles) | ForEach-Object { $_ -replace '\\', '/' } | Where-Object { $_ -match "^sdk/.+/" }
+  # remove files that target any specific package
+  foreach ($package in $allPackageProperties) {
+    $serviceFiles = $serviceFiles | Where-Object { "$RepoRoot/$_" -notmatch "^$($package.DirectoryPath)/" }
+  }
+
+  $affectedServiceDirectories = $serviceFiles | ForEach-Object { $_ -replace '^sdk/(.+?)/.*', '$1' } | Sort-Object -Unique
+
+  $affectedPackages = $allPackageProperties | Where-Object { $affectedServiceDirectories -contains $_.ServiceDirectory }
+
+  $additionalPackages += $affectedPackages | Where-Object { $packagesWithChanges -notcontains $_ }
+
+  # if the change affected no packags, e.g. eng/common change, we use core and template for validation
+  if ($additionalPackages.Length -eq 0) {
+    $additionalPackages += $allPackageProperties | Where-Object { $_.Name -eq "azure_core" -or $_.Name -eq "azure_template" }
+  }
+
+  return $additionalPackages
```

Comment:
You need to ensure that the packages you're adding don't already exist. I believe that the package-properties resolution should take care of it for you, but for correctness sake.

Reply:
That's the where-object statement on line 135.

additionalPackages will only be packages not already in packagesWithChanges

Reply:
Moved the declaration down to make it clearer